elastic/kibana
# [EDR Workflows] Automated Actions in more rule types

### Description: 
This PR enhances Kibana's rule-based alerting by adding support for Automated Actions in ESQL, EQL and NewTerms rules, thus extending the flexibility and automation capabilities of the platform.

⚠️ Currently the UI part is hidden behind a feature flag: `rulesResponseActionsTemporaryFlagEnabled`

Currently just supported in the above + query rules - because in order to use Response Action we need `agent.id` be included in the `alert`.

### Changes Introduced
**Support for ESQL Rules:** 

<img width="1525" alt="ESQL Rules Screenshot" src="https://github.com/elastic/kibana/assets/16632552/49d15b78-08f2-44b2-a345-6d6c2398d083">

**Support for EQL Rules:**

<img width="1524" alt="EQL Rules Screenshot" src="https://github.com/elastic/kibana/assets/16632552/7a2d27d8-4bc5-428d-8da0-c0a79707877e">


**Support for NewTerms Rules:**

<img width="2045" alt="Zrzut ekranu 2024-09-10 o 10 09 00" src="https://github.com/user-attachments/assets/107e5839-efac-4552-abc7-7a745af1179e">

### How to Test:

1. Add Osquery and Defend Workflows integrations to policy with enrolled host.
2. Navigate to the rules section in Kibana.
3. Create a new rule using the ESQL | EQL | New Terms query type.
4. Define the conditions for the rule.
5. Set up Automated Response Actions.
6. Verify that the action is triggered when the query conditions are met.
